src/vsr/replica.zig

```diff
@@ -6992,8 +6952,71 @@ pub fn ReplicaType(
                 assert(self.grid.read_faulty_queue.count == 1);
 
                 self.grid_repair_message_timeout.start();
-                self.on_grid_repair_message_timeout();
+                self.send_request_blocks();
+                self.grid.on_next_tick(
+                    on_grid_read_fault_next_tick,
+                    &self.grid_read_fault_next_tick,
+                );
```

Comment:
Hm, this:

* requests blocks immediately
* and also schedules requesting blocks on the next tick

We should do only the latter, right?

Reply:
Good catch; I left the `send_request_blocks` call in by mistake. :facepalm: